youtube_dl/extractor/common.py

```diff
@@ -780,9 +787,9 @@ def _formats_key(f):
                 if proto is None:
                     proto = compat_urllib_parse_urlparse(f.get('url', '')).scheme
 
-                preference = 0 if proto in ['http', 'https'] else -0.1
-                if f.get('ext') in ['f4f', 'f4m']:  # Not yet supported
```

Comment:
Yes, but we should use `self._extract_f4m_formats` instead, which already sets the `ext` field